### PR TITLE
Add active learning module

### PR DESCRIPTION
**Pull request by noamzbr**
temp for reviewing

### CODE REVIEW

File: soundbay/active_learning.py [hunk 1]
```diff
@@ -0,0 +1,165 @@
...
+        return chunk_end_sec
+    elif chunk_end_sec - chunk_start_sec > chunk_actual_size:
+        chunk_end_sec = chunk_start_sec + chunk_actual_size
+        return chunk_end_sec
+    else:
+        print('chunk size error!')
+        exit()
+
+
+def add_chunk_id_per_recording(df: pd.DataFrame, recording_id: str, chunk_len_in_seconds: int = 300) -> pd.DataFrame:
```
Comment (noamzbr) [inline soundbay/active_learning.py:60]: I'm not sure I understood why this whole thing existis. 
My best guess currently is that "segment" are the lengths the file was cut into for inference, while chunks are a different segmentation of the file with a different time interval used to mark "checks" that are then prioritized for active learning.

Question 1 in that case - why not prioritize the original segments? 
Thought 2 - seems like there should be an easier way to do that (if this is indeed what is done...), I mean if both lenghs are fixed theres a simple mathematical conversion. I'm probably missunderstanding something though.

File: soundbay/active_learning.py [hunk 2]
```diff
@@ -0,0 +1,165 @@
...
+
+
+def get_ranked_segments_from_inference_dir(inference_dir: str, chunk_len_in_seconds: int,
+                                           gain_threshold: float) -> pd.DataFrame:
+    df = load_inference_results_from_dir(inference_dir)
+    for recording_id in df.recording.unique():
+        df = add_chunk_id_per_recording(df, recording_id, chunk_len_in_seconds)
+    df = compute_potential_gain_per_segment(df)
+    ranked_chunks = df.groupby('chunk_id').apply(
+        lambda x: n_high_priority_segments(x, threshold=gain_threshold)).sort_values(
```
Comment (noamzbr) [inline soundbay/active_learning.py:145]: Indeed, why use a threshold and not n (n being "number of segments which we are willing to annotate" or such).

File: soundbay/active_learning.py [hunk 3]
```diff
@@ -0,0 +1,165 @@
...
+def load_inference_results_from_dir(inference_dir: str) -> pd.DataFrame:
+    df_all_recordings_inference = pd.DataFrame(
+        columns=['segment_id', 'recording', 'class0_prob', 'class1_prob', 'segment_start_sec', 'segment_end_sec'])
+
+    for filename in os.listdir(inference_dir):
+        df_one_recording_inference = create_inference_df_for_one_recording(filename, inference_dir)
+        df_all_recordings_inference = pd.concat([df_all_recordings_inference, df_one_recording_inference],
+                                                ignore_index=True)
+
+    df_all_recordings_inference['segment_end_sec'] = df_all_recordings_inference['segment_start_sec'] + 1
```
[the remark this answers is not in the document]
Comment (noa-weiss) [reply soundbay/active_learning.py:21]: You mean line 18 and 21? Or different ones?
Could you point out what's unclear, so I can explain it better?

[the remark this answers is not in the document]
Comment (noa-weiss) [reply soundbay/active_learning.py:21]: Resolved (online).

File: soundbay/active_learning.py [hunk 4]
```diff
@@ -0,0 +1,165 @@
...
+    return recording_name
+
+
+def load_inference_results_from_dir(inference_dir: str) -> pd.DataFrame:
+    df_all_recordings_inference = pd.DataFrame(
+        columns=['segment_id', 'recording', 'class0_prob', 'class1_prob', 'segment_start_sec', 'segment_end_sec'])
+
+    for filename in os.listdir(inference_dir):
+        df_one_recording_inference = create_inference_df_for_one_recording(filename, inference_dir)
+        df_all_recordings_inference = pd.concat([df_all_recordings_inference, df_one_recording_inference],
```
[the remark this answers is not in the document]
Comment (noa-weiss) [reply soundbay/active_learning.py:19]: Interesting. Efficiency is so not a problem with this, though, that I don't think it's worth changing.

File: soundbay/active_learning.py [hunk 5]
```diff
@@ -0,0 +1,165 @@
+import argparse
+import numpy as np
+import pandas as pd
+import os
+
+
+def get_recording_name_from_inference_file_name(inference_file_name):
+    recording_name = inference_file_name.split('\\')[-1].split('.')[0].split('-')[-1]
```
[the remark this answers is not in the document]
Comment (noa-weiss) [reply soundbay/active_learning.py:8]: So, I looked into it, and I don't think it is specific to a certain dataset - it's supposed to work for the naming format of our model's inference files.
I'll check it against a couple more, just to be sure.

File: soundbay/active_learning.py [hunk 6]
```diff
@@ -0,0 +1,194 @@
...
+    return recording_name
+
+
+def load_inference_results_from_dir(inference_dir: str, segment_length_in_seconds: int) -> pd.DataFrame:
+    df_all_recordings_inference = pd.DataFrame(
+        columns=['segment_id', 'recording', 'class0_prob', 'class1_prob', 'segment_start_sec', 'segment_end_sec'])
```
Comment (yogevmatalon) [inline soundbay/active_learning.py:33]: I'm assuming you developed a solution for the binary case, and it does not support multiclassification (>2 labels). maybe raise an error for trying to load a file with a "class2_prob" column?

File: soundbay/active_learning.py (hunk 5 above)
[the remark this answers is not in the document]
Comment (noa-weiss) [reply soundbay/active_learning.py:27]: Yup, works just fine.

File: soundbay/active_learning.py (hunk 6 above)
Comment (noa-weiss) [reply soundbay/active_learning.py:33]: That's a good idea :)

**Review by noa-weiss — APPROVED** [review]

**Pull request merged**